guava/src/com/google/common/io/LineReader.java

```diff
@@ -84,4 +94,43 @@ public String readLine() throws IOException {
     }
     return lines.poll();
   }
+
+    /**
+     * Returns a {@code Stream}, the elements of which are lines read from
+     * this {@code LineReader}.
+     *
+     * @return a {@code Stream<String>} providing the lines of text
+     *         described by this {@code LineReader}
+     */
+    public Stream<String> lines()
+    {
+        Iterator<String> iterator = new Iterator<String>() {
+            String nextLine = "";
+            @Override
+            public boolean hasNext() {
+                if (!nextLine.equals("")) {
+                    return true;
+                } else {
+                    try {
+                        nextLine = Optional.ofNullable(readLine()).orElse("");
+                        return (!nextLine.equals(""));
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                }
+            }
+            @Override
+            public String next() {
+                if (!nextLine.equals("") || hasNext()) {
```

Comment:
The `!nextLine.equals("")` condition should be already covered by the hasNext() right?